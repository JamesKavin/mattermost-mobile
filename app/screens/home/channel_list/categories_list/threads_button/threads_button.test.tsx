// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';

import {renderWithIntlAndTheme} from '@test/intl-test-helper';

import Threads from './threads_button';

test('Threads Component should match snapshot', () => {
    const {toJSON} = renderWithIntlAndTheme(
        <Threads
            currentChannelId='someChannelId'
            unreadsAndMentions={{
                unreads: 0,
                mentions: 0,
            }}
        />,
    );

    expect(toJSON()).toMatchSnapshot();
});
