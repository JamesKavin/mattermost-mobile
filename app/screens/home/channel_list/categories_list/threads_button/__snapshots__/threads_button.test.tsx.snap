// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Thread item in the channel list Threads Component should match snapshot 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID="channel_list.threads.button"
>
  <View
    style={
      Object {
        "marginLeft": -18,
        "marginRight": -20,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "minHeight": 40,
            "paddingHorizontal": 20,
          },
          undefined,
        ]
      }
    >
      <Icon
        name="message-text-outline"
        style={
          Array [
            Object {
              "color": "rgba(255,255,255,0.5)",
              "fontSize": 24,
            },
            undefined,
          ]
        }
      />
      <Text
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "fontFamily": "OpenSans",
              "fontSize": 16,
              "fontWeight": "400",
              "lineHeight": 24,
            },
            Object {
              "color": "rgba(255,255,255,0.72)",
              "marginTop": -1,
              "paddingLeft": 12,
              "paddingRight": 20,
            },
            undefined,
            undefined,
          ]
        }
      >
        Threads
      </Text>
    </View>
  </View>
</View>
`;

exports[`Thread item in the channel list Threads Component should match snapshot with only unreads filter 1`] = `null`;

exports[`Thread item in the channel list Threads Component should match snapshot, groupUnreadsSeparately false, always show 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID="channel_list.threads.button"
>
  <View
    style={
      Object {
        "marginLeft": -18,
        "marginRight": -20,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "minHeight": 40,
            "paddingHorizontal": 20,
          },
          undefined,
        ]
      }
    >
      <Icon
        name="message-text-outline"
        style={
          Array [
            Object {
              "color": "rgba(255,255,255,0.5)",
              "fontSize": 24,
            },
            undefined,
          ]
        }
      />
      <Text
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "fontFamily": "OpenSans",
              "fontSize": 16,
              "fontWeight": "400",
              "lineHeight": 24,
            },
            Object {
              "color": "rgba(255,255,255,0.72)",
              "marginTop": -1,
              "paddingLeft": 12,
              "paddingRight": 20,
            },
            undefined,
            undefined,
          ]
        }
      >
        Threads
      </Text>
    </View>
  </View>
</View>
`;
